content/_guides/how-to-define-envirament-variables-for-components.md

```diff
@@ -0,0 +1,88 @@
+---
+title: How to define environment variables for components
+layout: article
+section: Environment Variables
+order: 1
+category: environment-variables
+---
+
+## Description
+
+Environment variables are an essential part of the {{site.data.tenant.name}} platform environment. Apart from the ever growing [list of system-wide variables](/references/envirament-variables-available-during-component-execution), it is possible and rather encouraged, to define custom environment variables.
+
+## Why use environment variables?
+
+Let us consider the following use case when a component is developed for a CRM, Shop or ERP. If this particular component is intended to be used only for one specific client then many values can be hard-coded inside the component.
+
+What if the component is developed to work for different clients? All the hard-coded values for the different clients need to be rewritten again and again. This will introduce an unnecessary amount of overhead work and be the potential source of errors.
+
+
+
+Here are a couple of examples where the environment variables are implemented.
+
+## Example1: Environment Variables as OAuth Client keys
+
+Connecting an external service with {{site.data.tenant.name}} platform is usually handled through the security credentials which are used to establish a communication between our platform and the external service.
+
+In the cases when a system-wide deployed component is used on our main tenant (https://app.elastic.io) then login/password pair or API Keys are usually enough. The establishment of a secure connection with the external servers is handled by our system using our own OAuth Client keys and OAuth Apps created by us. For this case no further configuration is necessary.
```

Comment:
When a system-wide deployed component is used, the login/password pair or API Keys are usually enough. The establishment of a secure connection with the external servers is handled by our system using our own OAuth Client keys and OAuth Apps created by us. For this case no further configuration is necessary.